daffodil-tdml-processor/src/main/scala/org/apache/daffodil/processor/tdml/TDMLInfosetInputter.scala

```diff
@@ -81,7 +82,12 @@ class TDMLInfosetInputter(
         case _ if (firstVersion eq null) => ""
         // the json infoset inputter maintains CRLF/CR, but XML converts CRLF/CR to
         // LF. So if this is Json, then we want the CRLF/CR converted to LF
-        case jsonii: JsonInfosetInputter => firstVersion.replaceAll("(\r\n|\r)", "\n")
+        // Furthermore, the json infoset inputter maintains Unicode PUA characters, the XML
+        // infoset inputters have to remap some characters from the PUA back to the
+        // XML-illegal characters. For sake of comparison with other XML infoset inputters
+        // we also remap AS IF it were XML.
+        case jsonii: JsonInfosetInputter =>
+          XMLUtils.remapPUAToXMLIllegalCharacters(firstVersion.replaceAll("(\r\n|\r)", "\n"))
```

Comment:
Can we remove the replaceAll? I think remapPUAToXMLIllegalCharacters also remaps CRLF/CR to LF?

Which makes sense. XML infoset outputters remap characters so we need to do the same for JSON for TDML tests to verify it's all the same.